requirements.txt

```diff
@@ -3,3 +3,4 @@ matplotlib
 pillow
 sphinx
 nose
+nbformat
```

Comment:
> nbformat pulls in traitlets, ipython_genutils, jsonchema, and jupyter_core.

They are indeed a lot of dependencies, it's too much

> It seems like a big cost, given that writing a notebook is not much more than writing a json file. How difficult would it be to have our own code for writing such a file?

Not difficult at all. I started with nbformat, because that is the library for the notebook file and it provides their utilities. Now, I just used python standard json library. The file is not too ugly after enforcing indentation on save(Now that anyone might be willing to read it). It does not come out nicely ordered as IPython seems to prefer. But It the file is a valid JSON file and IPython seems to open it without problems. Upon saving it from the notebook it does recover the nice ordering.